src/storage/src/hummock/event_handler/hummock_event_handler.rs

```diff
@@ -208,16 +212,24 @@ impl HummockEventHandler {
         newly_uploaded_sstables: Vec<StagingSstableInfo>,
     ) {
         if !newly_uploaded_sstables.is_empty() {
-            let mut read_version_guard = self.read_version.write();
             newly_uploaded_sstables
                 .into_iter()
                 // Take rev because newer data come first in `newly_uploaded_sstables` but we apply
                 // older data first
                 .rev()
                 .for_each(|staging_sstable_info| {
-                    read_version_guard.update(VersionUpdate::Staging(StagingData::Sst(
-                        staging_sstable_info,
-                    )))
+                    let read_version_mapping_guard = self.read_version_mapping.write();
```

Comment:
Can we have a common method for iterating all read versions, such as `for_each_read_version`.